worker/upgradeseries/worker.go

```diff
@@ -352,6 +352,14 @@ func (w *upgradeSeriesWorker) pinLeaders() (err error) {
 
 	results, err := w.PinMachineApplications()
 	if err != nil {
+		// If pin machine applications method return not implemented because it's
+		// utilising the legacy leases store, then we should display the warning
+		// in the log and return out. Unpinning leaders should be safe as that
+		// should be considered a no-op
+		if errors.IsNotImplemented(errors.Cause(err)) {
```

Comment:
IsNotImplemented already handles the Cause portion, so you don't have to do that here.

Reply:
So during testing, we don't actually get a NotImplemented error, so this doesn't currently work. Is there anyway to check if the error is not implemented after it's crossed the API boundary?

Reply:
It's ok I found out how to do it :smile: